docs/usecases.md

```diff
@@ -0,0 +1,126 @@
+4. [Different rules for different users per host](#multi-environment)
+5. [Drop-in-Replacement](#drop-in-replacement)
+
+!!! tip
+
+    We're planning to also implement a [Docker](https://github.com/engity-com/bifroest/issues/11) and a [Kubernetes](https://github.com/engity-com/bifroest/issues/12) environment. This will create much more use-cases, soon. 🤠
+
+## Off-board users within 15 minutes of the organization {:id=offboard}
+
+### Problem
+
+1. Assume you're part of an organization.
+2. Assume this organization has more than _just_ 10 people who might be able to access SSH resources.
+3. Assume you've to off-board an employee, now.
+4. Assume you should ensure that this employee is unable to do harm to the organization, as the machines are crucial for the organization's technical security.
+
+In cases of SSH servers, this often results in running through all servers and either:
+
+* Change the passwords,
+* Remove dedicated users,
+* Remove user's public keys (if the command really tell you who it is 🤯),
+* or change the [Ansible](https://www.ansible.com/) or [Puppet](https://www.puppet.com/) configuration apply it at every machine.
+
+How this should be done within 15 minutes (not days or weeks)?<br>
+How do you ensure you really removed this user everywhere?
+
+### Solution
+
+#### Don't
+1. Have users installed on the systems itself.
+2. Share passwords of shared users or even the `root` user.
+3. Have user's public keys stored at shared users or even the `root` user.
+
+#### Do
+Use the [OpenID Connect authorization](reference/authorization/oidc.md).
+
+As the users always authorized by your [Identity Provider (IdP)](https://openid.net/developers/how-connect-works/), and this is always evaluated when someone tries to access the service via SSH, it will also immediately reject the authorization to this service.
+
+No need to access any of these services directly to remove/de-authorize these users.
+
+If the [environments are configured accordingly](reference/environment/index.md) (which is the default) all user's files and processes will be removed/killed automatically, too.
+
+## On-board users within 15 minutes in the organization {:id=onboard}
+
+This is quite similar to [Off-board users within 15 minutes of the organization](#offboard), but obviously reverse.
+
+### Problem
+
+1. Assume you're part of an organization.
+2. Assume this organization has more than _just_ 10 people who might be able to access SSH resources.
+3. Assume you've to on-board an employee, now.
+4. Assume you should ensure that this employee is ab to have access to all services, now.
+
+In cases of SSH servers, this often results in running through all servers and either:
+
+* Share the server shared-user passwords,
+* Add user's public key to a shared user,
+* Add a dedicated user (with password or authorized key),
+* or change the [Ansible](https://www.ansible.com/) or [Puppet](https://www.puppet.com/) configuration apply it at every machine.
```

Comment:
or changing the [Ansible](https://www.ansible.com/) or [Puppet](https://www.puppet.com/) configuration and apply it at every machine.